Compilation failure with gcc 4.4.7

Compilation of 4.0.0alpha fails with gcc 4.4.7 (gcc (GCC) 4.4.7 20120313 (Red Hat 4.4.7-17)):

`libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I.. -O2 -DNDEBUG -I../ccutil -I../viewer -I/usr/local/include/leptonica -g -O2 -MT simddetect.lo -MD -MP -MF .deps/simddetect.Tpo -c simddetect.cpp  -fPIC -DPIC -o .libs/simddetect.o
In file included from ../ccutil/genericvector.h:29,
                 from ../ccutil/params.h:25,
                 from ../ccutil/tprintf.h:23,
                 from simddetect.cpp:19:
../ccutil/helpers.h: In member function ‘void tesseract::TRand::set_seed(const std::string&)’:
../ccutil/helpers.h:50: error: ‘hash’ is not a member of ‘std’
`

Steps to reproduce: under an "old" OS (centos6 for example) follow the autogen&configure&make compilation steps and see the compilation error.

Rootcause: std hash was not officially supported in gcc 4.4, only in the TechnicalReport "tr1" namespace i.e. : std::tr1::hash<T> ... 

Possible fix: perhaps just enforcing a "recent" gcc at configure time ?

--- a/configure.ac
+++ b/configure.ac
@@ -368,14 +368,34 @@ AC_COMPILE_IFELSE(
   ]])
 ], [
      AC_MSG_RESULT(yes)
-     has_cpp11=yes
    ],
    [
      AC_MSG_RESULT(no)
-     has_cpp11=no
+     AC_MSG_ERROR([Your compiler does not have the necessary c++11 support! Cannot proceed.])
    ])
 AC_CHECK_FUNCS([snprintf],, [snprintfworks=yes])
 CXXFLAGS="$OLD_CXXFLAGS"
+
+
+# set c++11 support based on platform/compiler
+case "${host_os}" in
+  cygwin*)
+    CXXFLAGS="$CXXFLAGS -std=gnu++11"
+    ;;
+  *-darwin* | *-macos10*)
+     if test "x$CLANG" = "xyes"; then
+       CXXFLAGS="$CXXFLAGS -std=c++11 "
+       LDFLAGS="$LDFLAGS -stdlib=libc++"
+     else
+       CXXFLAGS="$CXXFLAGS -std=c++11"
+     fi
+    ;;
+  *)
+    # default
+    CXXFLAGS="$CXXFLAGS -std=c++11"
+    ;;
+esac
+
 
 # ----------------------------------------
 # Check for libraries
@@ -458,29 +478,6 @@ else
       CPPFLAGS="$CPPFLAGS $cairo_CFLAGS"
 fi
 
-# set c++11 support based on platform/compiler
-if test "x$has_cpp11" = "xyes"; then
-  case "${host_os}" in
-    cygwin*)
-      CXXFLAGS="$CXXFLAGS -std=gnu++11"
-      ;;
-    *-darwin* | *-macos10*)
-       if test "x$CLANG" = "xyes"; then
-         CXXFLAGS="$CXXFLAGS -std=c++11 "
-         LDFLAGS="$LDFLAGS -stdlib=libc++"
-       else
-         CXXFLAGS="$CXXFLAGS -std=c++11"
-       fi
-      ;;
-    *)
-      # default
-      CXXFLAGS="$CXXFLAGS -std=c++11"
-      ;;
-  esac
-else
-  AC_MSG_WARN([Training tools WILL NOT be built because of missing c++11 support.])
-  AM_CONDITIONAL([ENABLE_TRAINING], false)
-fi
 
 # ----------------------------------------
 # Final Tasks and Output

